### PR TITLE
[IMP]: add unit in form selection widget

### PR DESCRIPTION
**Pull request by maisim**
[FIX] hr_holidays_credit: plurals management in form selection widget
[IMP] hr_holidays_credit: Update french translation
[IMO] hr_holidays_credit: add messages on tests

### CODE REVIEW

File: hr_holidays_credit/models/hr_leave_type.py [hunk 1]
```diff
@@ -47,27 +47,28 @@ def name_get(self):
 
             extra = None
             if record.allocation_type != 'no' and context_employee_id:
-                if record.virtual_remaining_leaves >= 0:
+                amount = float_round(record.virtual_remaining_leaves, precision_digits=2) or 0.0
+                if(amount >= 0):
```
Comment (alexey-pelykh) [inline hr_holidays_credit/models/hr_leave_type.py]: `(` and `)` are not needed at all

File: hr_holidays_credit/models/hr_leave_type.py [hunk 2]
```diff
@@ -47,27 +47,28 @@ def name_get(self):
...
+                    if record.request_unit == 'day':
+                        extra = _('%g day available') % amount if abs(amount) <= 1 else _('%g days available') % amount
+                    elif record.request_unit == 'hour':
+                        extra = _('%g hour available') % amount if abs(amount) <= 1 else _('%g hours available') % amount
+                    else:
+                        extra = _('%g %s available') % (amount, record.request_unit)
+
                     if record.allow_credit:
-                        extra = _('%g available + credit')
+                            extra += " " + _('+ credit')
```
Comment (alexey-pelykh) [inline hr_holidays_credit/models/hr_leave_type.py]: Indent went very bad

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

**Comment by maisim** [comment]
Thanks for your review @alexey-pelykh

File: hr_holidays_credit/tests/test_hr_holidays_credit.py [hunk 3]
```diff
@@ -148,7 +148,8 @@ def test_6(self):
         name = leave_type.with_context(
             employee_id=employee.id,
         ).name_get()[0][1]
-        self.assertTrue('available + credit' in name)
+        self.assertIn('available', name, '%s not found in %s' % ('available', name))
```
Comment (alexey-pelykh) [inline hr_holidays_credit/tests/test_hr_holidays_credit.py]: What's the point of formatting the error message?

File: hr_holidays_credit/models/hr_leave_type.py [hunk 4]
```diff
@@ -47,27 +47,28 @@ def name_get(self):
...
-                        float_round(
-                            -record.virtual_remaining_leaves,
-                            precision_digits=2
-                        ) or 0.0,
-                    )
+                        extra = _('%g %s used in credit') % (amount, record.request_unit)
 
             if extra:
-                record_name = _('%(name)s (%(extra)s)') % {
+                record_name = '%(name)s (%(extra)s)' % {
```
Comment (alexey-pelykh) [inline hr_holidays_credit/models/hr_leave_type.py]: Why did you remove translation here? RTL languages kind-a have that backwards

File: hr_holidays_credit/tests/test_hr_holidays_credit.py (hunk 3 above)
Comment (maisim) [reply hr_holidays_credit/tests/test_hr_holidays_credit.py]: To know why test don't pass, So, I add messages to display compared values. I can remove it know but that's here... Let me know.
It was because my odoo test db default language was not English so name values was translated in the db default language (add -l or --load-language to command line does not help) I add to create a new one with english as default language

File: hr_holidays_credit/models/hr_leave_type.py (hunk 4 above)
Comment (maisim) [reply hr_holidays_credit/models/hr_leave_type.py]: To save white bears :angel:  
To me it's useless because it mean you have strings like : "Legal Leaves 2021 (20 days available)" in your po file, wich is very unlikely

Comment (alexey-pelykh) [reply hr_holidays_credit/models/hr_leave_type.py]: That's not how translation works in this case. Template is getting localized, not formatted string

Comment (maisim) [reply hr_holidays_credit/models/hr_leave_type.py]: I don't doubt what you're saying but I would like to understand. (I probably miss something)
In this case, for example,at line 70:
name is string containing "Legal Leaves 2021"
extra is string containing " (20 days available)"

record_name is string = name + " " + extra   = a string containing  "Legal Leaves 2021 (20 days available)"

So I don't understand how it can work in another way than having the full text "Legal Leaves 2021 (20 days available)" in the po file (or in the db)

Comment (alexey-pelykh) [reply hr_holidays_credit/models/hr_leave_type.py]: Pay attention to what's inside the `_()` and compare this

```
_('%(name)s (%(extra)s)') % {
    'name': 'A',
    'extra': 'B',
}
```

and 

```
_('%(name)s (%(extra)s)' % {
    'name': 'A',
    'extra': 'B',
})
```

File: hr_holidays_credit/tests/test_hr_holidays_credit.py (hunk 3 above)
Comment (alexey-pelykh) [reply hr_holidays_credit/tests/test_hr_holidays_credit.py]: I'm not sure if that's the correct way, to include localization in testing. I'd refer this to other @OCA/human-resources-maintainers

Comment (maisim) [reply hr_holidays_credit/tests/test_hr_holidays_credit.py]: The goal is not to test localization, it was to undesrtand why test failed has I had no tested values in logs

Comment (alexey-pelykh) [reply hr_holidays_credit/tests/test_hr_holidays_credit.py]: Runbot tests in en_US, so probably does not matter

File: hr_holidays_credit/models/hr_leave_type.py (hunk 4 above)
Comment (maisim) [reply hr_holidays_credit/models/hr_leave_type.py]: Ok, it allow to move vars in translations 
But in our case I think vars are not enough splitted to this use case.
Anyway I will undo this change.
Thanks for your time

File: hr_holidays_credit/tests/test_hr_holidays_credit.py (hunk 3 above)
Comment (maisim) [reply hr_holidays_credit/tests/test_hr_holidays_credit.py]: Yes but when I ran tests locally I can't see why the tests failed ( tests results didn't print variables content)
I then add messages to the tests which allowed me to see that the messages were translated (the reason why the failed tests) and to run the tests again on a basis with English as the default language.

Comment (maisim) [reply hr_holidays_credit/tests/test_hr_holidays_credit.py]: I removed custom messages in the last commit in the same time a had updated the tests

**Comment by maisim** [comment]
Can I merge commits (rebase) on my side or this repo allow squash merging ?

**Comment by alexey-pelykh** [comment]
You have to tidy up commit history, no squash merging

File: hr_holidays_credit/__manifest__.py [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
 {
     'name': 'Leave Credit',
-    'version': '12.0.1.0.0',
+    'version': '12.0.2.0.0',
```
Comment (alexey-pelykh) [inline hr_holidays_credit/__manifest__.py]: This is also not needed, auto-bump is in place, and this is not a major change.

File: hr_holidays_credit/models/hr_leave_type.py [hunk 6]
```diff
@@ -6,70 +6,86 @@
...
     )
     creditable_department_ids = fields.Many2many(
-        string='Creditable Departments',
-        comodel_name='hr.department',
+        string="Creditable Departments",
+        comodel_name="hr.department",
         help=(
-            'If set, limits credit allowance to employees of specified'
-            ' departments'
+            "If set, limits credit allowance to employees of specified" " departments"
```
Comment (alexey-pelykh) [inline hr_holidays_credit/models/hr_leave_type.py]: Clean up `" "` after autoformatting

File: hr_holidays_credit/models/hr_leave_type.py [hunk 7]
```diff
@@ -6,70 +6,86 @@
...
-                        ) or 0.0,
-                    )
-                elif record.allow_credit:
-                    extra = _('%g used in credit') % (
-                        float_round(
-                            -record.virtual_remaining_leaves,
-                            precision_digits=2
-                        ) or 0.0,
-                    )
+                        extra += " " + _("+ credit")
```
Comment (alexey-pelykh) [inline hr_holidays_credit/models/hr_leave_type.py]: Don't do this, it's incorrect for i18n to "construct" sentences. You have to localize either entire message or major part of it.

**Comment by maisim** [comment]
Hi @alexey-pelykh,
Can you have a look on this ?
Thank you in advance
